pkgs/regen.nix

```diff
@@ -0,0 +1,53 @@
+let
+  localLib = import ../lib.nix;
+  jemallocOverlay = self: super: {
+    # jemalloc has a bug that caused cardano-sl-db to fail to link (via
+    # rocksdb, which can use jemalloc).
+    # https://github.com/jemalloc/jemalloc/issues/937
+    # Using jemalloc 510 with the --disable-initial-exec-tls flag seems to
+    # fix it.
+    jemalloc = self.callPackage ../nix/jemalloc/jemalloc510.nix {};
+  };
+in
+{ system ? builtins.currentSystem
+, config ? {}
+, pkgs ? (import (localLib.fetchNixPkgs) { inherit system config; overlays = [ jemallocOverlay ]; })
+
+# Update this if you need a package version recently uploaded to hackage.
+# Any timestamp works.
+, hackageSnapshot ? "2018-07-17T09:58:14Z"
```

Comment:
what happens if we run it without `hackageSnapshot`? I'm just wondering, because the LTS is specified in `stack.yaml`. Is this if the default in nixpkgs doesn't have that LTS version in it?

Reply:
This timestamp is provided as an option to `stack2nix`. It needs to be there so that `stack2nix` has reproducible results (without it, latest hackage snapshot would be used).